target/linux/ath79/image/generic.mk

```diff
@@ -402,6 +402,27 @@ define Device/belkin_f9j1108-v2
 endef
 TARGET_DEVICES += belkin_f9j1108-v2
 
+define Device/belkin_f9k1115v2
+  SOC := qca9558
+  DEVICE_VENDOR := Belkin
+  DEVICE_MODEL := F9K1115V2 (AC1750 DB Wi-Fi)
+  IMAGE_SIZE := 14464k
+  DEVICE_PACKAGES += kmod-ath10k-ct ath10k-firmware-qca988x-ct kmod-usb2 \
+	kmod-usb3 kmod-usb-ledtrig-usbport
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x50000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+	lzma | uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | \
+	append-rootfs | pad-rootfs | check-size | \
+	edimax-headers eDiMaX F9K1115V2 | pad-to $$$$(BLOCKSIZE)
```

Comment:
That's a lot of duplicate definition, please create a shared one (there should be enough examples in the file).

Reply:
OK, I will consolidate this.
do you prefer `Device/belkin_f9x_v2_common` or `Device/belkin_edimax_headers` (or something else)?

Reply:
The former. You might even drop the common, just using Device/belkin_f9x_v2 or similar.

Experience tells that being too general doesn't help here (unless you have other devices to join as well, I haven't checked).